keanu-python/tests/test_sampling.py

```diff
@@ -53,3 +56,45 @@ def test_down_sample_interval(net):
 
     expected_num_samples = draws / down_sample_interval
     assert all(len(vertex_samples) == expected_num_samples for vertex_id, vertex_samples in samples.items())
+
+
+@pytest.mark.parametrize("algo", [
+    ("metropolis"),
+    ("hamiltonian")
+])
+def test_iter_is_correct_size(algo, net):
+    draws = 10
+    samples = kn.iterate_samples(net=net, sample_from=net.get_latent_vertices(), algo=algo, draws=draws, down_sample_interval=1)
+    assert sum(1 for i in samples) == 10
```

Comment:
Similarly here - you don't actually need the sample, so using "_" will (marginally) speed up the code.

Reply:
For readability, I think we should just make it a list and check its length?

Reply:
I do wish they had a length function for this!  I think just wrapping the sum thing in a function would be better - this pretty much is the idiomatic way to do it after all...